packages/connect-common/src/storage.ts

```diff
@@ -0,0 +1,50 @@
+// https://github.com/trezor/connect/blob/develop/src/js/storage/index.js
+const storageVersion = 1;
+const storageName = `storage_v${storageVersion}`;
+
+export const BROWSER_KEY = 'browser';
+export const PERMISSIONS_KEY = 'permissions';
+export const TRACKING_ENABLED = 'tracking_enabled';
+export const TRACKING_ID = 'tracking_id';
+
+interface Permission {
+    type: string;
+    origin?: string;
+    device?: string;
+}
+
+export interface Store {
+    [BROWSER_KEY]?: boolean;
+    [PERMISSIONS_KEY]?: Permission[];
+    [TRACKING_ENABLED]?: boolean;
+    [TRACKING_ID]?: string;
+}
+
+type GetNewStateCallback = (currentState: Store) => Store;
+
+let memoryStorage: Store = {};
+
+const getPermanentStorage = () => {
+    const ls = localStorage.getItem(storageName);
+
+    return ls ? JSON.parse(ls) : {};
+};
+
+export const save = (getNewState: GetNewStateCallback, temporary = false) => {
```

Comment:
What are the benefits of storing the whole state in one object instead of using separate keys? The downside is that it's more prone to conflict - if you are trying to save two state changes at the same time it's possible that only the last change will be saved, right? Or maybe not? – So far with the localstorage it could be fine, since it is synchronous?

cc @mroz22 ?

Reply:
- local storage is sync, so it shouldn't be a problem, at the moment. 
- for the future, this is part of bigger discussion about some persistent storage finally done well. but this is another topic. 
- benefit is imho that it is easier to work with and mantain, I don't need to always create a new key, import it everywhere etc.

Reply:
you still have to define new key at least in type, but never mind, since it's sync, we could keep it